compiler/ObjectFileEmitter/ObjectFileEmitter.cc

```diff
@@ -404,15 +406,26 @@ bool ObjectFileEmitter::run(spdlog::logger &logger, llvm::LLVMContext &lctx, uni
     addModulePasses(pm);
     pm.add(Passes::createRemoveUnnecessaryHashDupsPass());
     // print lowered IR
-    auto nameOptl = ((string)dir) + "/" + (string)objectName + ".lll";
-    llvm::raw_fd_ostream lllFile(nameOptl, ec1, llvm::sys::fs::F_Text);
-    auto *printLowered = llvm::createPrintModulePass(lllFile, "");
+    llvm::ModulePass *printLowered = nullptr;
+    // We put the ostream in an std::optional declared outside the "if", because unfortunately `createPrintModulePass`
+    // doesn't take ownership of it.
```

Comment:
Thanks for the comment here! I can totally see myself refactoring this to move the file handle into the conditional :)